onnxruntime/core/providers/cpu/ml/label_encoder.cc

```diff
@@ -55,168 +52,267 @@ Status LabelEncoder::Compute(OpKernelContext* context) const {
+        .TypeConstraint("T2", std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::string>()}),
     LabelEncoder_2<std::int64_t, std::string>)
 
 template <>
 void LabelEncoder_2<std::int64_t, std::string>::InitializeSomeFields(const OpKernelInfo& info) {
-  _key_field_name = "keys_int64s";
-  _value_field_name = "values_strings";
-  info.GetAttrOrDefault<std::string>("default_string", &_default_value, std::string("_Unused"));
+  key_field_name_ = "keys_int64s";
+  value_field_name_ = "values_strings";
+  info.GetAttrOrDefault<std::string>("default_string", &default_value_, std::string("_Unused"));
 };
 
-ONNX_CPU_OPERATOR_TYPED_ML_KERNEL(
-    LabelEncoder,
-    2,
-    string_int64,
-    KernelDefBuilder().TypeConstraint("T1",
-                                      std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::string>()})
-        .TypeConstraint("T2",
-                        std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::int64_t>()}),
+ONNX_CPU_OPERATOR_VERSIONED_TYPED_ML_KERNEL(
+    LabelEncoder, 2, 3, string_int64,
+    KernelDefBuilder()
+        .TypeConstraint("T1", std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::string>()})
+        .TypeConstraint("T2", std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::int64_t>()}),
     LabelEncoder_2<std::string, std::int64_t>)
 
 template <>
 void LabelEncoder_2<std::string, std::int64_t>::InitializeSomeFields(const OpKernelInfo& info) {
-  _key_field_name = "keys_strings";
-  _value_field_name = "values_int64s";
-  info.GetAttrOrDefault<std::int64_t>("default_int64", &_default_value, (std::int64_t)-1);
+  key_field_name_ = "keys_strings";
+  value_field_name_ = "values_int64s";
+  info.GetAttrOrDefault<std::int64_t>("default_int64", &default_value_, (std::int64_t)-1);
 };
 
-ONNX_CPU_OPERATOR_TYPED_ML_KERNEL(
-    LabelEncoder,
-    2,
-    int64_int64,
-    KernelDefBuilder().TypeConstraint("T1",
-                                      std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::int64_t>()})
-        .TypeConstraint("T2",
-                        std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::int64_t>()}),
+ONNX_CPU_OPERATOR_VERSIONED_TYPED_ML_KERNEL(
+    LabelEncoder, 2, 3, int64_int64,
+    KernelDefBuilder()
+        .TypeConstraint("T1", std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::int64_t>()})
+        .TypeConstraint("T2", std::vector<MLDataType>{DataTypeImpl::GetTensorType<std::int64_t>()}),
     LabelEncoder_2<std::int64_t, std::int64_t>)
 
 template <>
 void LabelEncoder_2<std::int64_t, std::int64_t>::InitializeSomeFields(const OpKernelInfo& info) {
-  _key_field_name = "keys_int64s";
-  _value_field_name = "values_int64s";
-  info.GetAttrOrDefault<std::int64_t>("default_int64", &_default_value, (std::int64_t)-1);
+  key_field_name_ = "keys_int64s";
+  value_field_name_ = "values_int64s";
+  info.GetAttrOrDefault<std::int64_t>("default_int64", &default_value_, (std::int64_t)-1);
```

Comment:
>std::int64_t)- [](http://example.com/codeflow?start=73&length=14)

No C casts

Reply:
Again, this was existing code but good point.